core/src/main/resources/swagger/api-docs.yaml

```diff
@@ -1394,6 +1394,42 @@ paths:
         500:
           $ref: '#/components/responses/RawlsInternalError'
       x-codegen-request-body-name: newEntityNameJson
+  /api/workspaces/{workspaceNamespace}/{workspaceName}/entities/renameEntityType:
```

Comment:
This signature works for me so I'm happy to approve as-is, but I wanted to offer an alternative signature that is more RESTful and could avoid confusing bugs in the future:

PATCH:  `/api/workspaces/{workspaceNamespace}/{workspaceName}/entities/{entityType}`

And then a body that takes the new entity type name.

I think the ship already sailed on this long ago, but the API paths for entities are getting to be a bit crowded. I believe that REST principles state that the request verb and body should determine the action to perform, and the API path should determine the path to the resource that should be acted upon.

By adding actions to the API paths, it creates ambiguity, i.e. is a particular segment in the path telling us where to find the resource, or what action to perform? An example of the problem that this ambiguity creates can be found in our current APIs. See these two endpoints, one for getting a TSV for a certain entity type, and another for getting a single entity:

GET `/api/workspaces/{workspaceNamespace}/{workspaceName}/entities/{entityType}/tsv`
GET `/api/workspaces/{workspaceNamespace}/{workspaceName}/entities/{entityType}/{entityName}`

Because the /tsv path is resolved first, it actually makes it impossible to call the second GET endpoint if your entity is named "tsv". No user has ever reported this bug, but it's still an issue and fixing it would be a bit of work. Preferably, the getTsv endpoint would have been something like `GET /api/workspaces/{workspaceNamespace}/{workspaceName}/entities/{entityType}&format=tsv`

Reply:
You raise a number of valid points, Matt.  `PATCH` would be the right HTTP verb for this case according to Roy Fielding. Muddling the REST standard by putting `renameEntityType` in the url does at least help with the practical problem of avoiding url collisions in the future, but your suggested pattern would not collide with any existing urls at the moment.  I'm going to try making the PATCH and url pattern change and see if that's straightforward.

Reply:
I went through the exercise of making the changes you suggest, @MatthewBemis .  You can see those changes over in #1699  Changing over to `PATCH` is quite straightforward, but while doing so I did notice/remember that switching to `PATCH` is at odds with the existing rename entity end point.  Is it worth being more RESTful with a new end point if that new end point is not consistent with an already published end point?  I'm not sure, but I'd lean towards probably not. 

Making the url change such that the entity type being changed is a path parameter was a bit less straightforward due to swagger.  Swaggers complains about this:     

PATCH: /api/workspaces/{workspaceNamespace}/{workspaceName}/entities/{entityType}
GET: /api/workspaces/{workspaceNamespace}/{workspaceName}/entities/{entityType}

It says that we have duplicate keys despite the urls associating with distinct verbs so I had to change the url to: 

PATCH: /api/workspaces/{workspaceNamespace}/{workspaceName}/entities/{entityTypeToRename}

(and add a new path param to the swagger).  I don't have a strong opinion on this, but if I had to pick I'd say consistency within Rawls' API (i.e. making rename entity type similar to rename entity) is a stronger impetus than adhering to REST standards.  @jordansafer @davidangb  what do you think?

Reply:
For POST vs PATCH, I'm in favor of consistency. I think we could make a future ticket to clean up the APIs by making PATCH endpoints and deprecating the POST endpoints. For the URL change, I do think it's a good point that we're overloading with /entities/{entityType} and /entities/renameEntityType. I would vote to avoid the overload, and go with adding a path param. But that seems to require the PATCH change, in which case this could all be work for a future ticket.

Reply:
I also see that we have a `PATCH` method for updating an entity's attributes so admittedly the consistency argument is not so cut and dry.  Rawls is consistently inconsistent perhaps, but I would say renaming and entity and renaming an entity type are closer relatives than renaming an entity type and updating an entity's attributes.

Reply:
I think that API consistency is a compelling argument. And to be clear, I think this is a larger issue that goes far beyond the scope of this PR.

It's a bit of a moot point now, but the duplicate key error in the Swagger can be fixed by nesting both method types under the same API path. See [this example](https://github.com/broadinstitute/rawls/blob/develop/core/src/main/resources/swagger/api-docs.yaml#L145) where there is both a put and a delete listed underneath.

Reply:
Cool, thank you for the swagger tip! I'm going to merge this as is for the sake of consistency with the rename entity end point with the acknowledgement that this API will surprise and frustrate a REST adherent :face_exhaling: